Presentr/KeyboardTranslation.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  KeyboardTranslation.swift
+//  Presentr
+//
+//  Created by Aaron Satterfield on 7/15/16.
+//  Copyright © 2016 danielozano. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public enum KeyboardTranslationType {
+    case None, MoveUp, Compress, StickToTop
+    
+    /**
+     Calculates the correct frame for the keyboard translation type.
+     
+     - parameter keyboardFrame: The UIKeyboardFrameEndUserInfoKey CGRect Value of the Keyboard
+     - parameter presentedFrame: The frame of the presented controller that may need to be translated.
+     - returns: CGRect representing the new frame of the presented view.
+     */
+    public func getTranslationFrame(keyboardFrame : CGRect, presentedFrame : CGRect) -> CGRect {
+        let keyboardTop = UIScreen.mainScreen().bounds.height - keyboardFrame.size.height
+        let presentedViewBottom = presentedFrame.origin.y + presentedFrame.height + 20.0 // add a 20 pt buffer
+        let offset = presentedViewBottom - keyboardTop
+        switch self {
+        case .MoveUp:
+            if offset > 0.0 {
+                let frame = CGRectMake(presentedFrame.origin.x, presentedFrame.origin.y-offset, presentedFrame.size.width, presentedFrame.size.height)
+                return frame
+            }
+            return presentedFrame
+        case .Compress:
+            if offset > 0.0 {
+                let y = max(presentedFrame.origin.y-offset, 20.0)
+                let newHeight = y != 20.0 ? presentedFrame.size.height : keyboardTop - 40.0
+                let frame = CGRectMake(presentedFrame.origin.x, y, presentedFrame.size.width, newHeight)
```

Comment:
@aasatt Swift constructors are preferred over legacy convenience functions. Use CGRect and its initialisers instead of CGRectMake.